### DISCUSSION TITLE
Can Slint and Winit be used simultaneously?

### DISCUSSION THREAD

**Discussion by daydaystudylab** [slint-ui/slint#4391] [Q&A]
thread 'main' panicked at src\main.rs:11:31:                  
called `Result::unwrap()` on an `Err` value: RecreationAttempt

`use slint::{LogicalPosition, WindowPosition};
use winit::dpi::LogicalSize;
use winit::event_loop::EventLoop;
use winit::window::WindowBuilder;
slint::include_modules!();

fn main() {

    let app = App::new().unwrap();

    let ep = EventLoop::new().unwrap();
    println!("{:?}", ep);

    let window = WindowBuilder::new().build(&ep).unwrap();
    let scale = window.scale_factor();
    let size: LogicalSize<f32> = window.current_monitor().unwrap().size().to_logical(scale);
    println!("{:?}", size);
    let x = (size.width - 890.0) / 2.0;
    let y = (size.height - 624.0) / 2.0;

    let lp = LogicalPosition::new(x, y);
    app.window().set_position(WindowPosition::Logical(lp));
    app.run().unwrap();
}
`

**Comment by ogoffart** [comment]
Slint uses winit itself, so both can't use winit.

Using the private internal API, you can access the winit data scruture from the i-slint-backend-winit crate.
See https://docs.rs/i-slint-backend-winit/latest/i_slint_backend_winit/

with with_event_loop_window_target you can access the EventLoopWindowTarget needed to create new windows.
Unfortunately, you won't be able to handle events. (feature requested in https://github.com/slint-ui/slint/issues/3066 )

**Reply by daydaystudylab** [reply]
thank you so much
